integrations/gitlab/gitlab_integration/git_integration.py

```diff
@@ -4,14 +4,17 @@
     FILE_PROPERTY_PREFIX,
     SEARCH_PROPERTY_PREFIX,
 )
-from gitlab_integration.gitlab_service import PROJECTS_CACHE_KEY
 from loguru import logger
 from pydantic import Field
 from gitlab.v4.objects import Project
 
 from port_ocean.context.event import event
 from port_ocean.core.handlers import JQEntityProcessor
-from port_ocean.core.handlers.port_app_config.models import PortAppConfig
+from port_ocean.core.handlers.port_app_config.models import (
+    PortAppConfig,
+)
+
+PROJECTS_CACHE_KEY = "__cache_all_projects"
```

Comment:
that shouldnt be in this file

i understand why you did it but there are other ways to solve such an issue. You did it becuae you couldnt import the GitlabConfiguration class to cast the configuration to it

if you are trying to import something that is only a type without having a circular error you can do it by checking if the code is in typechcking mode with this code

```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from gitlab_integration.git_integration import GitlabPortAppConfig
    
```

Then you can use the class by wrapping its usage with `"`

it will look something like this

`typing.cast(obj, "GitlabPortAppConfig")`


You can see something similar in `context/resource.py`